File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String ret = "";
        int mode = 0;
        for(int i = 0; i < code.length(); i++){
            char c = code.charAt(i);
            if(c == '1'){
                if(mode == 0) mode = 1;
                else mode = 0;
            } else {
                if(mode == 0 && i % 2 == 0) ret += c;
                else if(mode == 1 && i % 2 != 0) ret += c;
            }
        }
        if(ret.length() == 0) ret = "EMPTY";
        return ret;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    ret, mode  = '', 0
    for i,x in enumerate(code):
        if mode == 0:
            if x != '1':
                if i % 2  == 0:
                    ret += x
            else:
                mode = 1
        else:
            if x != '1':
                if i % 2 == 1:
                    ret += x
            else:
                mode = 0
    return ret if ret else "EMPTY"
                
            
            
    
"""
mode:0 => 1아니면 idx 짝수일 때 ret에 code[idx] 추가, 1이며 mode=0
mode:1 => 1아니면, idx 홀 ret에 code[idx] 추가, 1이면 mode= 1

"""